sql-plugin/src/main/scala/com/nvidia/spark/rapids/RmmRapidsRetryIterator.scala

```diff
@@ -872,3 +900,8 @@ object RetryStateTracker {
 
   def clearCurThreadRetrying(): Unit = localIsRetrying.remove()
 }
+
+object SplitReason extends Enumeration {
+  type SplitReason = Value
+  val GPU_OOM, CPU_OOM, OTHER = Value
```

Comment:
Since we know the exact reason as "ColumnSizeOverflow", why name it as OTHER ?

Reply:
this is in case in the future we'll have other special exceptions besides ColumnSizeOverflow

Reply:
Good point, then we can have more SplitReason values to map them.

Reply:
I'm not sure if we want so many states. To me, "Other" is a good enough category to include all kinds of future exceptions. If we as a troubleshooter wants to know what is happening, we may need to take a look at the log printed by https://github.com/NVIDIA/spark-rapids/pull/12352/files#diff-06c3cea1c7b935018328547859836a21d3b935b3c80da6e05f09f455abe043a6R697

Reply:
I am OK with OTHER if other guys agreee with it.